lib/ansible/modules/cloud/cloudstack/cs_image_store.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2019, Patryk Cichy @PatTheSilent
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: cs_image_store
+
+short_description: Manages CloudStack Image Stores.
+
+version_added: "2.8"
+
+description:
+  - "Deploy, remove, recreate CloudStack Image Stores."
+
+options:
+  url:
+    description:
+      - The URL for the Image Store.
+      - Required when I(state=present).
+    type: str
+  name:
+    description:
+      - The ID of the Image Store. Required when deleting a Image Store.
+    required: true
+    type: str
+  zone:
+    description:
+      - The Zone name for the Image Store.
+    required: true
+    type: str
+  state:
+    description:
+      - Stage of the Image Store
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  provider:
+    description:
+      - The image store provider name. Required when creating a new Image Store
+    type: str
+  force_recreate:
+    description:
+      - Set to C(True) if you're changing an existing Image Store.
+      - This will force the recreation of the Image Store.
+      - Recreation might fail if there are snapshots present on the Image Store. Delete them before running the recreation.
+    type: bool
+    default: no
+
+extends_documentation_fragment: cloudstack
+
+author:
+  - Patryk Cichy (@PatTheSilent)
+'''
+
+EXAMPLES = '''
+- name: Add a Image Store (NFS)
+  cs_image_store:
+    zone: zone-01
+    name: nfs-01
+    url: nfs://192.168.21.16/exports/secondary
```

Comment:
The param `provider` is required when creating.

Reply:
Fell in my own trap... Done.